vsphere/resource_vsphere_virtual_disk_test.go

```diff
@@ -106,96 +123,189 @@ func testAccVSphereVirtualDiskExists(name string) resource.TestCheckFunc {
 
 		dc, err := finder.Datacenter(context.TODO(), rs.Primary.Attributes["datacenter"])
 		if err != nil {
-			return fmt.Errorf("error %s", err)
+			return err
 		}
 		finder = finder.SetDatacenter(dc)
 
 		ds, err := finder.Datastore(context.TODO(), rs.Primary.Attributes["datastore"])
 		if err != nil {
-			return fmt.Errorf("error %s", err)
+			return err
 		}
 
 		_, err = ds.Stat(context.TODO(), rs.Primary.Attributes["vmdk_path"])
 		if err != nil {
-			return fmt.Errorf("error %s", err)
+			if testAccCheckVSphereVirtualDiskIsFileNotFoundError(err) && expected == false {
+				// Expected missing
+				return nil
+			}
+			return err
+		}
+		if !expected {
+			return fmt.Errorf("expected virtual disk %s to be missing", rs.Primary.ID)
 		}
-
 		return nil
 	}
 }
 
-func testAccCheckVSphereVirtualDiskDestroy(s *terraform.State) error {
-	log.Printf("[FINDME] test Destroy")
-	client := testAccProvider.Meta().(*VSphereClient).vimClient
-	finder := find.NewFinder(client.Client, true)
+func testAccCheckVSphereVirtualDiskIsFileNotFoundError(err error) bool {
+	if strings.HasPrefix(err.Error(), "cannot stat") && strings.HasSuffix(err.Error(), "No such file") {
+		return true
+	}
+	return false
+}
 
-	for _, rs := range s.RootModule().Resources {
-		if rs.Type != "vsphere_virtual_disk" {
-			continue
-		}
+func testAccCheckVSphereVirtuaDiskConfig_basic(rName string) string {
+	return fmt.Sprintf(`
+variable "datacenter" {
+  default = "%s"
+}
 
-		dc, err := finder.Datacenter(context.TODO(), rs.Primary.Attributes["datacenter"])
-		if err != nil {
-			return fmt.Errorf("error %s", err)
-		}
+variable "datastore" {
+  default = "%s"
+}
 
-		finder = finder.SetDatacenter(dc)
+variable "rstring" {
+  default = "%s"
+}
 
-		ds, err := finder.Datastore(context.TODO(), rs.Primary.Attributes["datastore"])
-		if err != nil {
-			return fmt.Errorf("error %s", err)
-		}
+data "vsphere_datacenter" "dc" {
+  name = "${var.datacenter}"
+}
 
-		_, err = ds.Stat(context.TODO(), rs.Primary.Attributes["vmdk_path"])
-		if err == nil {
-			return fmt.Errorf("error %s", err)
-		}
+data "vsphere_datastore" "ds" {
+  name          = "${var.datastore}"
+  datacenter_id = "${data.vsphere_datacenter.dc.id}"
+}
 
-		// Clean up the test parent directory created for evaluation of the attribute "create_directories"
-		if rs.Primary.Attributes["create_directories"] == "true" {
-			fm := object.NewFileManager(client.Client)
-			vmdkPath := rs.Primary.Attributes["vmdk_path"]
-			directoryPathIndex := strings.LastIndex(vmdkPath, "/")
-			path := vmdkPath[0:directoryPathIndex]
+resource "vsphere_virtual_disk" "foo" {
+  size         = 1
+  vmdk_path    = "tfTestDisk-${var.rstring}.vmdk"
+  adapter_type = "lsiLogic"
+  type         = "thin"
+  datacenter   = "${data.vsphere_datacenter.dc.name}"
+  datastore    = "${data.vsphere_datastore.ds.name}"
+}
+`,
+		os.Getenv("VSPHERE_DATACENTER"),
```

Comment:
Nit: It might be more clear to reference these in the acceptance test functions themselves so its near the `PreCheck` declaration and then require the variables be passed into the wrapping function for the `fmt.Sprintf()` so its not possible to forget them, but that's generally how we do it in the AWS provider. Either way is probably fine!

Reply:
I actually like your way more as it would make the config functions more reusable. At this point, I'm going to just stick with this way just to keep it consistent with the other vsphere tests.